password_enhancements.module

```diff
@@ -0,0 +1,471 @@
+      'required' => [],
+    ];
+    $valid_required = 0;
+    $valid_optional = 0;
+    $plugin_manager = Drupal::service('plugin.manager.password_constraint');
+    foreach ($constraints as $constraint) {
+      /** @var \Drupal\password_enhancements\Plugin\PasswordConstraintPluginInterface $plugin_instance */
+      $plugin_instance = $plugin_manager->createInstance($constraint->getType(), $constraint->getConfiguration());
+      $is_required = $constraint->isRequired();
+      $is_valid = $plugin_instance->validate($password);
+
+      if ($is_valid && $is_required) {
+        $valid_required++;
+      }
+      elseif ($is_valid) {
+        $valid_optional++;
+      }
+      else {
+        $error_message_collection[$is_required ? 'required' : 'optional'][] = $plugin_instance->getErrorMessage();
+      }
+    }
+
+    // Collect error messages.
+    $error_messages = [];
+    if (($valid_required + $valid_optional) < $minimum_required_constraints) {
+      $error_messages = array_merge($error_message_collection['required'], $error_message_collection['optional']);
+    }
+    elseif ($valid_required === 0 && $error_message_collection['required'] === [] && $valid_optional < $minimum_required_constraints) {
+      $error_messages = &$error_message_collection['optional'];
+    }
+    elseif ($error_message_collection['required'] !== []) {
+      $error_messages = &$error_message_collection['required'];
+    }
+
+    // Set form error and show the error message if there is any.
+    if ($error_messages !== []) {
+      $messages = [
+        '#theme' => 'password_enhancements_error_messages',
+        '#title' => t("The given password doesn't meet some requirements:"),
+        '#messages' => $error_messages,
+        '#attributes' => [
+          'class' => ['password-error-messages'],
+        ],
+        '#title_attributes' => [
+          'class' => ['title', 'password-error-message'],
+        ],
+      ];
+      $form_state->setError($form['account']['pass'], Drupal::service('renderer')->render($messages));
+    }
+  }
+}
+
+/**
+ * Submit callback for password_enhancements_form_user_register_form_alter().
+ *
+ * This callback runs only if the password field was set to always require and
+ * the verify_email set to true.
+ *
+ * @param array $form
+ *   The form.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current form state.
+ */
+function password_enhancements_form_user_register_form_alter_password_required_submit(array &$form, FormStateInterface $form_state): void {
+  /** @var \Drupal\user\UserInterface $user */
+  if (!empty($form['account']['pass'])) {
+    /** @var \Drupal\Core\Entity\EntityFormInterface $form_object */
+    $form_object = $form_state->getFormObject();
+    $user = $form_object->getEntity();
+
+    try {
+      $user->setPassword($form_state->getUserInput()['pass']['pass1'])->save();
+    }
+    catch (Exception $e) {
+      Drupal::service('logger.password_enhancements')->exception($e);
```

Comment:
Well, it would be better if I could see from the first sentence on the logs UI (from the log entry overviews) what has failed exactly and where. If you log whatever exceptions here I have to check all log entries to find the source of why XY's user's password has not been saved - and I can only see from the backtrace that ZW exception was thrown here but the backtrace is not visible on the overview page.

Reply:
Actually not much is printed out in the log list, only 56 characters are displayed from the message.

If we see errors regarding the module then those should be fixed/investigated either way, but anyway the first part of the actual error message (exception message) is visible in the list and that should be enough.

Reply:
Added subject argument.